### ISSUE TITLE
[Grid] Clarification for functions in grids

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2897]
### Provide a general summary of the issue here

The grids can have multiple actions, which might not be apparently clear to users, and are not necessarily apparent from some of the elements. Many actions should be clarified to ensure all users are able to understand the controls without trial and error.

<img width="1463" height="391" alt="Grid with 2 rows selected, one f" src="https://github.com/user-attachments/assets/669ea554-52e2-4329-9aa8-d08cd73060ed" />

In the examples there's a Selection with filtering variant, which has the following functions, which all generate somewhat different outcomes
	• Select all elements from the table
	• Sort the table by columns
	• Filter column based on text input or a select element
	• Select a single or multiple rows
	• Cancel the selection using the toolbar (this can have multiple different actions, at least on some types of grid)

Some of these directly manipulate how the table is arranged or its contents, and some of these are actions that act upon the table rows. Visually these functions have very distinct, but programmatically (i.e., to screen reader users) the functions have a learning curve.

Clarifying these actions before the user enters the table would help users understand the functions before a trial and error phase.

### 🤔 Expected Behavior?

Actions are clear before starting to manipulate the table

### 😯 Current Behavior

Actions require a lot of trial and error for those who do not have good technical skills or are using assistive technology and/or keyboard

### 💁 Possible Solution

Add information to the table that offers guidance for all users.

E.g. for the selection with filtering:
- Column one has a checkbox for each row, which selects the row. You can select multiple rows. The first row has a checkbox, which selects all rows. 
- After selection a toolbar will appear at the end of the table named "Actions for selected rows". There you can select actions for the selected rows.
- First row has table headers and buttons for sorting the table columns.
- Second row has filters, which will filter the contents of a column automatically based on input or selection.

[One way to achieve this is using a `<details>` tag like in my example](https://magsu.github.io/draganddrop_wcag257/)

### 🖥️ Steps to Reproduce

- 

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
